### PR TITLE
Handled possible unclear IndexError in ConversableAgent.last_message method

### PR DESCRIPTION
**Pull request by jeezrick**
Handled possible unclear IndexError in ConversableAgent.last_message method with more clear error message and added test in test_conversable_agent.py.

**Description of the Changes:**

I have implemented a safeguard in the `last_message` method to handle the scenario where the provided agent is not present in the keys of the messages. Previously, this situation could lead to a confusing "IndexError" due to an attempt to access an invalid index.

**Details:**

- Added a conditional check to ensure that the provided agent exists in the keys before attempting to access messages.
- If the agent is not found in the keys, the method now raises KeyError, and provides clear error message.

**Testing:**

I have added a test in `test_conversable_agent.py` to verify that the `last_message` method behaves as expected in this scenario. The test confirms that the method does not raise an "IndexError" when an agent not in the keys is provided. See https://github.com/jeezrick/autogen/blob/50db138af2772027ed711ab50b63a6ead99f06f6/test/agentchat/test_conversable_agent.py#L209C2-L209C2

**Note to Reviewers:**

Please review the changes and ensure that the modifications to the `last_message` method improve its robustness and prevent potential index errors in the specified scenario.

## Checks

- [ ] I've included any doc changes needed for https://microsoft.github.io/autogen/. See https://microsoft.github.io/autogen/docs/Contribute#documentation to build and test documentation locally.
- [ √] I've added tests (if relevant) corresponding to the changes introduced in this PR.
- [ √] I've made sure all auto checks have passed.

### CODE REVIEW

File: test/agentchat/test_conversable_agent.py [hunk 1]
```diff
@@ -242,7 +245,7 @@ async def test_a_generate_reply_raises_on_messages_and_sender_none(conversable_a
 
 
 if __name__ == "__main__":
-    test_trigger()
+    # test_trigger()
     # test_context()
     # test_max_consecutive_auto_reply()
-    # test_conversable_agent(pytest.monkeypatch)
+    test_conversable_agent()
```
Comment (sonichi) [inline test/agentchat/test_conversable_agent.py:251]: Is this line of change necessary?

**Review by sonichi — COMMENTED** [review]
Thanks. I have one minor comment.

File: test/agentchat/test_conversable_agent.py (hunk 1 above)
Comment (jeezrick) [reply test/agentchat/test_conversable_agent.py:251]: Thank you for your response. The reason I made this modification is that I want to test the `test_conversable_agent` function independently. However, running it with `pytest.monkeypatch` seems to be causing issues, and the function itself doesn't actually utilize any method form `monkeypatch` fixture(it's all commented out). Could you please help me understand why calling this function, when includeing pytest.monkeypatch, is not working? I appreciate your patience!

Comment (jeezrick) [reply test/agentchat/test_conversable_agent.py:251]: I can unchange this line.

Comment (sonichi) [reply test/agentchat/test_conversable_agent.py:251]: I tried it and you are right.

**Review by sonichi — APPROVED** [review]

**Comment by sonichi** [comment]
Please fix the code formatting issue https://github.com/microsoft/autogen/actions/runs/6835094216/job/18588498007?pr=622
Then we can merge.

**Comment by jeezrick** [comment]
> Please fix the code formatting issue https://github.com/microsoft/autogen/actions/runs/6835094216/job/18588498007?pr=622 Then we can merge.
Code formatting issues have been addressed. Are there any additional concerns or suggestions? This is my first time contributing to an open-source project, so I'm still learning the process and would appreciate any guidance or feedback.

**Pull request merged**